src/wheel/cli/unpack.py

```diff
@@ -14,10 +16,18 @@ def unpack(path: str, dest: str = ".") -> None:
     :param path: The path to the wheel.
     :param dest: Destination directory (default to current directory).
     """
+    umask = os.umask(0)
+    os.umask(umask)
     with WheelFile(path) as wf:
         namever = wf.parsed_filename.group("namever")
         destination = Path(dest) / namever
         print(f"Unpacking to: {destination}...", end="", flush=True)
-        wf.extractall(destination)
+        for zinfo in wf.filelist:
+            extracted_path = destination / zinfo.filename
```

Comment:
`zinfo` already has the filename path, so this is duplicating the path. `a/b/c.txt` -> `a/b/a/b/c.txt`.

Reply:
No, that makes no sense, as `destination` remains the same, only `zinfo.filename` keeps changing. I tested extracting `Django-3.2.16-py3-none-any.whl` (which happened to be lying around) with both this branch and `main`, and I didn't see any difference.

Reply:
You are setting destination with the filename, so now it has both. I think you want `wf.extract(zinfo, destination)` here. Extract extracts "a/b.txt" from argument 1 to argument 2, keeping the original relative path. Flatting this requires workarounds, see https://stackoverflow.com/a/47632134 for example. Are you sure you were using this branch?

```
tree Django-4.1.7/Django-4.1.7.dist-info
Django-4.1.7/Django-4.1.7.dist-info
├── AUTHORS
│   └── Django-4.1.7.dist-info
│       └── AUTHORS
├── LICENSE
│   └── Django-4.1.7.dist-info
│       └── LICENSE
├── LICENSE.python
│   └── Django-4.1.7.dist-info
│       └── LICENSE.python
├── METADATA
│   └── Django-4.1.7.dist-info
│       └── METADATA
├── RECORD
│   └── Django-4.1.7.dist-info
│       └── RECORD
├── WHEEL
│   └── Django-4.1.7.dist-info
│       └── WHEEL
├── entry_points.txt
│   └── Django-4.1.7.dist-info
│       └── entry_points.txt
└── top_level.txt
    └── Django-4.1.7.dist-info
        └── top_level.tx
```

Reply:
However, this only affects the non-package files, like `.dist-info`. The package files (for both packages) seem fine.

Reply:
On a deeper look, I saw the problem.

Reply:
I basically pushed almost the exact same fix right as you posted that comment :smile: